docs/proposals/deletions_object_storage.md

```diff
@@ -0,0 +1,86 @@
+*   **GDPR:** Masking data and eventual deletion is expected.
+*   **Deletions to sustain user requirements:** Let’s assume the user has some data which leads to some unexpected results or causes performance degradation (due to high cardinality) and the user wants to restore the previous data set-up for obtaining the desired results. In this scenario, the user would want to send a request to mask the data for the time being as there isn’t a high priority requirement to delete the data but eventually during the compaction the user can expect the data to be deleted by the compactor not leading to any major performance issues.
+*   **Helps achieving series based retention (e.g rule aggregation) [#903](https://github.com/thanos-io/thanos/issues/903).**
+
+
+## Goals
+
+
+
+*   Unblock users and allow series deletion in the object storage using tombstones.
+*   Allowing undeletes for a default time duration and extending it as per the user requirements.
+*   Dealing with already downsampled and to be downsampled blocks with tombstones.
+*   Performing all the above operations at admin level.
+
+
+# Proposed Approach
+
+
+
+*   We propose to implement deletions using the tombstones approach.
+*   A user is expected to enter the following details for performing deletions:
+    *   **external labels**
+    *   **start timestamp**
+    *   **end timestamp** (start and end timestamps of the series data the user expects to be deleted)
+    *   **maximum waiting duration for performing deletions** where a default value is considered if explicitly not specified by the user. (only after this time passes the deletions are performed by the compactor in the next compaction cycle)
+*   The details are entered via a deletions API (good to have a web UI interface) and they are processed by the compactor to create a tombstone file, if there’s a match for the details entered. Afterwards the tombstone file is uploaded to object storage making it accessible to other components.
+*   If the data with the requested details couldn’t be found in the storage an error message is returned back as a response to the request.
+*   Store Gateway masks the series on processing the tombstones from the object storage.
+*   **Perspectives to deal with Downsampling of blocks having tombstones:**
+    *   **Block with tombstones and max duration to perform deletes passed:** The compactor should first check the maximum duration to perform deletions for that block and if the proposed maxtime passed the deletions are first performed and then the downsampling occurs.
+    *   **Block with tombstones and max duration still hasn’t passed:** Perform compaction.
+    *   **Performing deletes on already compacted blocks:** Have a threshold to perform deletions on the compacted blocks (Prometheus has 5%)
```

Comment:
Could we elaborate more on what is this threshold?

Reply:
This threshold is something that Prometheus does as well. We need to check for the exact value and maybe should want to link here?

Reply:
For now we want to start with the 5% that Prometheus has. 
We want to follow up on that threshold on IRC/Matrix and ask why it was chosen. In the future, we can always be smarter about this.

Reply:
Adding this to the Prometheus community meeting, happening next week :)